src/io.ts

```diff
@@ -48,37 +73,39 @@ export async function generateTodoFiles(
   lintResults: LintResult[],
   filePath?: string
 ): Promise<string> {
-  const todoDir: string = await ensureTodoDir(baseDir);
-
-  const existing: Map<string, TodoData> = await readTodoFiles(todoDir, filePath);
-
+  const todoStorageDir: string = await ensureTodoDir(baseDir);
+  const existing: Map<string, TodoData> = await readTodos(todoStorageDir, filePath);
   const [add, remove] = await getTodoBatches(buildTodoData(lintResults), existing);
 
-  await _generateFiles(baseDir, add, remove);
+  await _generateFiles(todoStorageDir, add, remove);
 
-  return todoDir;
+  return todoStorageDir;
 }
 
 /**
  * Reads all todo files in the .lint-todo directory.
  *
- * @param todoDir The .lint-todo storage directory.
+ * @param todoStorageDir The .lint-todo storage directory.
  * @param filePath? The absolute file path of the file to return todo items for.
  */
-export async function readTodoFiles(
-  todoDir: string,
-  filePath?: string
+export async function readTodos(
+  todoStorageDir: string,
+  filesDirOrPath?: string
 ): Promise<Map<string, TodoData>> {
-  const fileNames = await readdir(todoDir);
+  if (filesDirOrPath) {
+    todoStorageDir = join(todoStorageDir, todoDirFor(filesDirOrPath));
+  }
 
+  const fileNames = await globby(todoStorageDir);
```

Comment:
I'd much prefer to avoid globby if we can. I don't see what it gives us here since we aren't using globs...

```js
function getTodoFiles(todoDir) {
  let todos = [];
  let todoFolders = fs.readdirSync(todoDir);
  
  for (let todoFolder of todoFolders) {
    let fileNames = fs.readdirSync(todoFolder).map(name => path.join(todoDir, todoFolder, name);

    todos.push(...fileNames);
  }

  return todos;
}
```

Reply:
It allows us to grab all files from the subdirectories of all the files in one-pass. Why do you prefer to not use it?

Reply:
Not to be overly pedantic, but the code you wrote isn't doing the same thing. It's a simplification of what we're actually doing, which includes the need to support getting lists of files under a single directory (for a single `filePath`), and creating a map of the `FilePath` and `Todo`. 🤷

Reply:
My code above is very close to what is need. I'm happy to push a commit with the changes if the snippet doesn't make sense out of context.

Reply:
No need. Updated!